runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt

```diff
@@ -1,101 +1,248 @@
+     */
+    public fun add(key: String, value: String?) {
+        content[key] = JsonPrimitive(value)
+    }
 
-    internal val content: MutableMap<String, JsonElement> = linkedMapOf()
+    internal fun build(): JsonObject = JsonObject(content)
 
     /**
      * Adds given [value] to the current [JsonObject] with [this] as a key.
      */
+    @Deprecated(message = infixToDeprecated, replaceWith = ReplaceWith("add(this, value)"))
     public infix fun String.to(value: JsonElement) {
-        require(content[this] == null) {"Key $this is already registered in builder"}
+        require(content[this] == null) { "Key $this is already registered in builder" }
         content[this] = value
     }
 
     /**
      * Adds given [value] as [JsonPrimitive] to the current [JsonObject] with [this] as a key.
      */
+    @Deprecated(message = infixToDeprecated, replaceWith = ReplaceWith("add(this, value)"))
     public infix fun String.to(value: Number?) {
-        require(content[this] == null) {"Key $this is already registered in builder"}
+        require(content[this] == null) { "Key $this is already registered in builder" }
         content[this] = JsonPrimitive(value)
     }
 
     /**
      * Adds given [value] as [JsonPrimitive] to the current [JsonObject] with [this] as a key.
      */
+    @Deprecated(message = infixToDeprecated, replaceWith = ReplaceWith("add(this, value)"))
     public infix fun String.to(value: Boolean?) {
-        require(content[this] == null) {"Key $this is already registered in builder"}
+        require(content[this] == null) { "Key $this is already registered in builder" }
         content[this] = JsonPrimitive(value)
     }
 
     /**
      * Adds given [value] as [JsonPrimitive] to the current [JsonObject] with [this] as a key.
      */
+    @Deprecated(message = infixToDeprecated, replaceWith = ReplaceWith("add(this, value)"))
     public infix fun String.to(value: String?) {
-        require(content[this] == null) {"Key $this is already registered in builder"}
+        require(content[this] == null) { "Key $this is already registered in builder" }
         content[this] = JsonPrimitive(value)
     }
 }
+
+/**
+ * DSL builder for a [JsonArray]. To create an instance of builder, use [buildJsonArray] build function.
+ */
+@JsonDslMarker
+public class JsonArrayBuilder internal constructor() {
+
+    private val content: MutableList<JsonElement> = mutableListOf()
+
+
+    /**
+     * Add the [JSON][JsonObject] produced by the [builderAction] function to a resulting array.
+     */
+    public fun addJson(builderAction: JsonObjectBuilder.() -> Unit) {
+        content += buildJson(builderAction)
+    }
+
+    /**
+     * Add the [JSON][JsonArray] produced by the [builderAction] function to a resulting array.
+     */
+    public fun addArray(builderAction: JsonArrayBuilder.() -> Unit) {
+        content.add(buildJsonArray(builderAction))
```

Comment:
Any particular reasons for using `add` here instead of `+=` like in other methods?

Reply:
Yes, `JsonArray` implements list, making `content += jsonArray` ambigous